doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,311 @@
+.. _ug_fw_update:
+
+Firmware updates
+################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This user guide provides an overview of firmware updates with |NCS|.
+
+.. _ug_fw_update_keys:
+
+Signature keys
+**************
+
+You can find information on the key types supported in the |NCS| in the :ref:`nRF Secure Immutable Bootloader <bootloader_signature_keys>` and :ref:`mcuboot:imgtool` documentation.
+
+Generating private keys
+=======================
+
+The most common ways, supported by the |NCS|, to generate private/public key pairs are the following:
+
+* :ref:`ug_fw_update_keys_openssl`
+* :ref:`ug_fw_update_keys_imgtool`
+* :ref:`ug_fw_update_keys_python`
```

Comment:
In ug_bootloader_adding.html#adding-a-custom-signature-key-file we state Python is the default method and list is as the first one. Any reason for listing Python as the last one here?

Reply:
@kyberdin

Reply:
@greg-fer I placed Python lowest on this list because I consider OpenSSL to be the de-factor standard tool for these sort of operations, and it supports a large number of keys, making it the most useful of the list to have recommended first.

The Python method here relies on the `ecdsa` package and is only relevant for creating private/public keys for ECDSA256, making it less important than `imgtool` or OpenSSL for this walkthrough. It is "default" in our build system in that `b0` will use it to autogenerate ECDSA keys when none are specified by the user. However, the docs introduced here highlight that developers should not rely on these for their projects and provide their own of whichever supported type they prefer.